depthai_nodes/ml/parsers/mlsd.py

```diff
@@ -45,13 +48,40 @@ def __init__(
         @param dist_thr: Distance threshold for merging lines.
         @type dist_thr: float
         """
-        dai.node.ThreadedHostNode.__init__(self)
-        self.input = self.createInput()
-        self.out = self.createOutput()
+        super().__init__()
+
         self.topk_n = topk_n
         self.score_thr = score_thr
         self.dist_thr = dist_thr
 
+    def build(
+        self,
+        head_config: Dict[str, Any],
+    ):
+        """Sets the head configuration for the parser.
+
+        Attributes
+        ----------
+        head_config : Dict
+            The head configuration for the parser.
+
+        Returns
+        -------
+        MLSDParser
+            Returns the parser object with the head configuration set.
+        """
+
+        output_layers = head_config["outputs"]
```

Comment:
Same as for palm detection, should save to `self.output_layers` and add it to initializer and add setMethods.

Reply:
Yeah I left it out intetionally. How is setting `self.output_layers` useful? We still need to get them by actual name e.g. `tpMap` and `heat`.

Reply:
You could do `self.output_layer_tpmap` and `self.output_layer_heat` and save them separately?

Reply:
Yeah IMO what @jkbmrz suggested makes sense. In the build we check if they are present based on the expected name and if they are we assign them automatically (self.output_layer_tpmap = "tpMap",...) else we error out saying they should use .setOutputLayerTpMap() instead. And in error message we tell them what are the expected output layer names that the model should have if they want to use build().

In run() we then just check if these are set and if they are we take them.